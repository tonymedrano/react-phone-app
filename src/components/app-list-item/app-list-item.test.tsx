import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom';

import AppListItem, { AppListItemProps } from './app-list-item';

const data:AppListItemProps  = {
    id: "1",
    name: "Galaxy S7",
    description: "Samsung Galaxy S7 Android smartphone.",
    image: "https://www.androidcentral.com/sites/androidcentral.com/files/styles/mediumplus/public/article_images/2016/03/galaxy-s7-13.jpg",
    color: "gray"
}

test('<AppListItem />', () => {
    const { asFragment } = render(<AppListItem { ...data }/>);
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot(asFragment());
});

test('<AppList /> name', () => {
    const comp = render(<AppListItem  { ...data }/>);
    const name =  comp.getByTestId('name');
    expect(name).toHaveTextContent('Galaxy S7');
});

test('<AppList /> color', () => {
    const comp = render(<AppListItem  { ...data }/>);
    const color =  comp.getByTestId('color');
    expect(color).toHaveTextContent('Color:');
});

test('<AppList /> image', () => {
    const comp = render(<AppListItem  { ...data }/>);
    const image =  comp.getByTestId('image');
    expect(image.src).toMatch('https://www.androidcentral.com/sites/androidcentral.com/files/styles/mediumplus/public/article_images/2016/03/galaxy-s7-13.jpg');
});