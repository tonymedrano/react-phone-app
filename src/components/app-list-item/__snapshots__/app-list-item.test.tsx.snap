// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppListItem /> 1`] = `Object {}`;
